### PR TITLE
GCW 3153 add constraints restriction to packages

### PR DESCRIPTION
**Pull request by abulsayyad123**
### Ticket Link: 

https://jira.crossroads.org.hk/browse/GCW-3153
### What does this PR do?

This PR adds constraints, saleable and comment field in packages creation and details page.

### CODE REVIEW

File: app/controllers/api/v1/packages_controller.rb [hunk 1]
```diff
@@ -356,7 +356,7 @@ def package_params
           :inventory_number, :item_id, :length, :location_id, :notes, :order_id,
           :package_type_id, :pallet_id, :pieces, :received_at, :saleable,
           :received_quantity, :rejected_at, :state, :state_event, :stockit_designated_on,
-          :stockit_id, :stockit_sent_on, :weight, :width, :favourite_image_id,
+          :stockit_id, :stockit_sent_on, :weight, :width, :favourite_image_id, :restriction_id,:comment,
```
Comment (steveyken) [inline app/controllers/api/v1/packages_controller.rb:359]: Add a space after ,

```suggestion
          :stockit_id, :stockit_sent_on, :weight, :width, :favourite_image_id, :restriction_id, :comment,
```

File: db/migrate/20200522105202_add_comment_to_packages.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class AddCommentToPackages < ActiveRecord::Migration
+  def change
+    add_column :packages, :comment, :text
```
Comment (steveyken) [inline db/migrate/20200522105202_add_comment_to_packages.rb:3]: We actually already have a 'notes' field. Talking with Matt, we don't need to add a separate field. Just use 'notes' and don't create a new comments field.

**Comment by abulsayyad123** [comment]
Hi @steveyken , sure i'll remove `comment` field. But right now `notes` is in sync with `description` field on stockit. Do I just need to sync `notes` on GC with `comment` on stockit (if we don't want `comment` in this PR) or keep `notes` with `description` (the existing implementation) ?
Let me know.

cc: @patrixr

**Comment by abulsayyad123** [comment]
Hi @steveyken , sure i'll remove `comment` field. But right now `notes` is in sync with `description` field on stockit. Do I just need to sync `notes` on GC with `comment` on stockit (if we don't want `comment` in this PR) or keep `notes` with `description` (the existing implementation) ?
Let me know.

cc: @patrixr

**Pull request merged**